src/main/java/com/nimbusds/jose/JWEObjectJSON.java

```diff
@@ -0,0 +1,497 @@
+/*
+ * nimbus-jose-jwt
+ *
+ * Copyright 2012-2021, Connect2id Ltd and contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.nimbusds.jose;
+
+import com.nimbusds.jose.util.Base64URL;
+import com.nimbusds.jose.util.JSONObjectUtils;
+import net.jcip.annotations.ThreadSafe;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * JSON Web Encryption (JWE) secured object.
+ *
+ * Provides <a href="https://datatracker.ietf.org/doc/html/rfc7516#section-7.2>JWE JSON Serialization</a>
+ *
+ * This class is thread-safe.
+ *
+ * @author Alexander Martynov
+ * @version 2021-08-17
+ */
+@ThreadSafe
+public class JWEObjectJSON extends JOSEObject implements JSONSerializable {
```

Comment:
I don't see any props of extending JWEObject.

I changed contract of encryption and decryption it's necessary because `JWEDecryptor` takes only encodedKey i don't want to extends this because in this case i need to support it for all decryptors